homebridge/thermostat.ts

```diff
@@ -0,0 +1,189 @@
+          // The current temperature is lower than the target temperature,
+          // and the thermostat is set to 'heat' or 'aux' (emergency heat), so the thermostat is heating
+          return Characteristic.CurrentHeatingCoolingState.HEAT
+        }
+        // The current temperature is either higher or lower than the target temperature,
+        // but the current thermostat mode would only to increase the difference,
+        // so the thermostat is neither heating nor cooling
+        return Characteristic.CurrentHeatingCoolingState.OFF
+      },
+    })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.TargetHeatingCoolingState,
+      serviceType: Service.Thermostat,
+      getValue: ({ mode }) => {
+        switch (mode) {
+          case 'off':
+            return Characteristic.TargetHeatingCoolingState.OFF
+          case 'heat':
+          case 'aux':
+            return Characteristic.TargetHeatingCoolingState.HEAT
+          case 'cool':
+            return Characteristic.TargetHeatingCoolingState.COOL
+        }
+      },
+      setValue: (targetHeatingCoolingState: number) => {
+        const mode: ThermostatMode | undefined = (():
+          | ThermostatMode
+          | undefined => {
+          switch (targetHeatingCoolingState) {
+            case Characteristic.TargetHeatingCoolingState.OFF:
+              return 'off'
+            case Characteristic.TargetHeatingCoolingState.HEAT:
+              return 'heat'
+            case Characteristic.TargetHeatingCoolingState.COOL:
+              return 'cool'
+            default:
+              return
+          }
+        })()
+        if (!mode) {
+          this.logger.error(
+            `Couldn’t match ${targetHeatingCoolingState} to a recognized mode string.`
+          )
+          return
+        }
+        this.logger.info(`Setting ${this.device.name} mode to ${mode}`)
+
+        return this.device.setInfo({ device: { v1: { mode } } })
+      },
+    })
+    // Only allow 'TargetHeatingCoolingState's which can be mapped to Ring modes
+    // Specifically, this omits .AUTO
+    this.getService(Service.Thermostat)
+      .getCharacteristic(Characteristic.TargetHeatingCoolingState)
+      .setProps({
+        validValues: [
+          Characteristic.TargetHeatingCoolingState.OFF,
+          Characteristic.TargetHeatingCoolingState.HEAT,
+          Characteristic.TargetHeatingCoolingState.COOL,
+        ],
+      })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.CurrentTemperature,
+      serviceType: Service.Thermostat,
+      getValue: async () => {
+        const temperature = await this.getTemperatureFromComponentDevices()
+        if (!temperature) {
+          return
+        }
+        // Documentation: https://developers.homebridge.io/#/characteristic/CurrentTemperature
+        // 'Characteristic.CurrentTemperature' supports 0.1 increments
+        return Math.round(temperature * 10) / 10
```

Comment:
Slightly different syntax option, if you think it's more clear:

```suggestion
        return Number(temperature.toFixed(1))
```

Reply:
I considered that, but `toFixed` throws a SyntaxError when `temperature` is an integer:
```
> 21.toFixed(1)
21.toFixed(1)
^^^

Uncaught SyntaxError: Invalid or unexpected token
```

I do think it’s more clear, and it looks like this slight variation works:
```JS
Number(Number(21).toFixed(1))
```

Reply:
Committed in https://github.com/dgreif/ring/pull/331/commits/52123c148ec0370a94c71168a9cc96520d15812b